modules/wallet/wallet_test.go

```diff
@@ -15,6 +16,16 @@ import (
 	"github.com/NebulousLabs/Sia/types"
 )
 
+var (
+	// Sometimes during testing, you get a statement
+	// `if err != expectedErr { return err }`. But if err == nil, then the
+	// calling function will think that the test has passed, even though the
+	// test failed. [This is a developer error, instead return
+	// errors.New("expecting..")] The new type of error catches when the
+	// developer has made a mistake and failed to catch a test suite failure.
+	errTestCompleted = errors.New("test complete")
```

Comment:
thought you might.

need a better solution then. There needs to be assurance that the test is making it all the way to the end, instead of failing silently.

If the dev uses errTestCompleted at all, it will be apparent that they forgot to use it correctly if they don't use it in both cases. You have to `return errTestCompleted` as well as run `if err != errTestCompleted` - if you do one but not the other:

1. The test fails because an unexpected error was returned, or because `nil` was returned.
2. The test fails silently because it would have failed silently anyway - `nil` was returned accidentally, and the dev forgot to  check `if err != errTestCompleted`.

If they forget to use it at all, then it's equivalent to having just not used it.

Therefore, using `errTestCompleted` is strictly less error-prone than not using `errTestCompleted`.

I recognize that it's unconventional. If you have an alternative proposal for making sure that a test doesn't accidentally return nil when it has actually failed, I'm happy to consider that. Passing the function a `testing.T` object is an unattractive alternative. Having the function call `panic` is also an unattractive alternative. Perhaps there's some construction using defer that will make sure the test completed - 

```go
defer func() {
    if !testFinished && err == nil {
        panic("test is failing silently!")
    }
}
```

If you are strongly opposed to `errTestFinished`, then I think I would be okay putting panics everywhere. I don't like the `testing.T` method of failing because it doesn't help you figure out what conditions caused the test to fail. `panic` on the other hand gives you  a full stack trace, which has the benefit of telling you both what line failed in the helper function and what line failed in the caller function.

Maybe `panic` is the best option.